xml/security-security-group-logging.xml

```diff
@@ -0,0 +1,153 @@
+  firewall driver. Under <literal>[securitygroup]</literal> in
+  <filename>~/openstack/my_cloud/config/neutron/ml2_conf.ini.j2</filename>,
+  change the firewall driver to <literal>firewall_driver = openvswitch</literal>.
+ </para>
+ <para>
+  Use the following steps to enable logging for
+  <literal>securitygroup</literal>.
+ </para>
+ <procedure>
+  <step>
+  <para>
+   Add log as a <literal>service_plugin</literal> in
+   <filename>~/openstack/my_cloud/config/neutron/neutron.conf.j2</filename>
+  </para>
+  <screen>service_plugins = {{ neutron_service_plugins }},log</screen>
+  </step>
+  <step>
+   <para>
+    Add the <literal>log</literal> extension in the <literal>agent</literal>
+    section of
+    <filename>~/openstack/my_cloud/config/neutron/ml2_conf.ini.j2</filename>.
+   </para>
+   <screen>[agent]
+extensions = log</screen>
+  </step>
+  <step>
+   <para>
+    Add the <literal>log</literal> extension in the <literal>agent</literal>
+    section of
+    <filename>~/openstack/my_cloud/config/neutron/openvswitch_agent.ini.j2</filename>. <emphasis
+    role="bold">If other extensions are configured (such as
+    <literal>qos</literal>), then the <literal>log</literal> extension must
+    added manually or the functionality of the other extension will
+    break.</emphasis>
+   </para>
+   <screen>[agent]
+extensions = log</screen>
+  </step>
+  <step>
+   <para>
+    Configure the <literal>network_log</literal> section in
+    <filename>~/openstack/my_cloud/config/neutron/openvswitch_agent.ini.j2</filename>. If
+    a custom file is configured to use for output logs, then log file rotation
+    must be done manually. Using a custom log file is optional. Set
+    <literal>rate_limit</literal> and <literal>burst_limit</literal> according
+    to the environment.
+   </para>
+   <screen>[network_log]
+rate_limit = 100
+burst_limit = 25
+local_output_log_base = /var/log/neutron/security_group.log</screen>
+  </step>
+  <step>
+   <para>
+    Commit changes to git.
+   </para>
+   <screen>&prompt.ardana;cd ~/openstack/ardana/ansible/
+&prompt.ardana;git add -A
+&prompt.ardana;git commit -m "Enable logging for security groups"</screen>
+  </step>
+  <step>
+   <para>
+    Run configuration processor and ready deployment playbooks.
+   </para>
+   <screen>&prompt.ardana;ansible-playbook -i hosts/localhost config-processor-run.yml
+&prompt.ardana;ansible-playbook -i hosts/localhost ready-deployment.yml</screen>
+  </step>
+  <step>
+   <para>
+    For a cloud that is already deployed, run the
+    <filename>neutron-reconfigure.yml</filename> playbook or follow cloud
+    deployment steps.
+   </para>
+   <screen>&prompt.ardana;cd ~/scratch/ansible/next/ardana/ansible/
+&prompt.ardana;ansible-playbook -i hosts/verb_hosts neutron-reconfigure.yml
+or
+&prompt.ardana;ansible-playbook -i hosts/verb_hosts site.yml</screen>
+  </step>
+ </procedure>
+ <note>
+  <para>
+   We recommend enabling logging for <literal>security_group</literal> and
```

Comment:
Anytime we use "\<literal\>" we need the text to be the same as the text on the system to which we are referring. So in the above line we should change "security_group" to "securitygroup" because here we are still talking about the deployment phase and so we are referring to the text in the ml2_conf.ini.j2 file. After deployment the text includes the underscore "security_group" because then it refers to the resource group which includes the underscore.  @italiahardik do you agree?